### ISSUE TITLE
Java implementation of .fai index creation

### ISSUE THREAD

**Issue by magicDGS** [samtools/htsjdk#830]
### Motivation

The `.fai` index from samtools is used in some parts of the code in HTSJDK, but there is no way to generate it within the library. Because it is well documented [here](http://www.htslib.org/doc/faidx.html) and it looks simple to implement, I think that HTSJDK could benefit for a java-based implementation of this indexing.

**Comment by tfenne** [comment]
I agree, this would be nice.  It would also be nice to have the capability to create and use the indices that are generated on bgzipped fasta files which samtools supports.

**Comment by dariober** [comment]
If any useful... I have the code to create `.fai` index from fasta (not bgzipped though). The code it's here: https://github.com/dariober/ASCIIGenome/tree/master/src/faidx

The tests are here: https://github.com/dariober/ASCIIGenome/blob/master/test/faidx/

Usage is pretty straightforward:

```
File fasta= new File("genome.fa"); 
new Faidx(fasta); // will create genome.fa.fai
```

Feel free to grab anything that maybe useful.

**Comment by magicDGS** [comment]
I've already started an implementation, but it may be useful to refine some parts of my code. Thank you very much, @dariober!

**Comment by magicDGS** [comment]
Closed by #842

**Comment by magicDGS** [comment]
@tfenne, I open a different issue regarding the capability to fully support compressed FASTA.
